mwang2000/eng-intern-assessment-react
# Completed Front End Challenge - Maggie Wang

https://github.com/mwang2000/eng-intern-assessment-react/assets/59301359/65ff3509-a0b0-410b-bc81-02ed1c8d54df

**I have completed the following project goals:**

StopWatch has the following functionality:
- Start the stopwatch to begin counting time.
- Stop the stopwatch to pause the timer.
- Displays Laps when a button is pressed.
- Reset the stopwatch to zero.

Ensure code quality:
- Write clean, well-structured, and maintainable code.
- Follow best practices and adhere to the React and TypeScript coding conventions.
- Pay attention to code readability, modularity, and performance.

Test your code:
- Run the application and test the stopwatch functionality to ensure it works correctly.
- Verify that the stopwatch starts, stops, resets, and records laps as expected.

Code documentation:
- Document your code by adding comments and explanatory notes where necessary.
- Provide clear explanations of the implemented functionality and any important details.

Version control:
- Use Git for version control. Commit your changes regularly and push them to a branch in your forked repository.

- I made a design choice to prevent the user from creating laps if the timer is sitting at 0, and only one lap can be created if the timer is paused to prevent user error. 
- I also chose to use a Utils file for some functions with logic that have a high chance of being reused.
- The tests were very confusing in both concept and operation, so I made a few modifications to the test file but still not all have passed even though the desired outcomes have been met, and I could not manually replicate several of the failed tests.
